chain/network/src/types.rs

```diff
@@ -773,11 +955,22 @@ pub enum NetworkRequests {
     /// Request given block headers.
     BlockHeadersRequest { hashes: Vec<CryptoHash>, peer_id: PeerId },
     /// Request state for given shard at given state root.
-    StateRequest { shard_id: ShardId, hash: CryptoHash, peer_id: PeerId },
+    StateRequest { shard_id: ShardId, hash: CryptoHash, account_id: AccountId },
     /// Ban given peer.
     BanPeer { peer_id: PeerId, ban_reason: ReasonForBan },
     /// Announce account
-    AnnounceAccount(AnnounceAccount),
+    AnnounceAccount(AnnounceAccount, bool),
```

Comment:
I already handled this case in `routing`. The only situation when we want to force announcement is when we are the originators. Also remove myself from routing table.